### PR TITLE
HPCC-25119 Add icu support to windows build action

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by ghalliday** [comment]
@Michael-Gardner @xwang2713  which of you would like to make sure that this doesn't cause problems with the current jenkins builds?

**Comment by Michael-Gardner** [comment]
@ghalliday putting it up on Jenkins to test now

**Review by Michael-Gardner — DISMISSED** [review]
Tested on our build environment. Everything compiles and links correctly when we update to icu67.

**Comment by Michael-Gardner** [comment]
@ghalliday tested and didn't find any issues. Should be ready for merge.  I have the packaging fix PR up at https://github.com/hpcc-systems/HPCC-Platform/pull/14477

**Comment by ghalliday** [comment]
@richardkchapman I have re-targeted to 8.0.  I think it is ready to merge - but worth waiting for smoketest etc.

File: rtl/include/eclhelper.hpp [hunk 1]
```diff
@@ -38,7 +38,7 @@ if the supplied pointer was not from the roxiemem heap. Usually an OwnedRoxieStr
 #include "unicode/utf.h"
 #else
 #ifdef _WIN32
-typedef wchar_t UChar;
+typedef char16_t UChar;
```
Comment (richardkchapman) [inline rtl/include/eclhelper.hpp:41]: Will this change any signatures of (non-virtual) API fucntions that might result in queries not being compatible? I suppose we probably don't mind so much if it's only windows though?

**Review by richardkchapman — APPROVED** [review]

File: rtl/include/eclhelper.hpp (hunk 1 above)
Comment (ghalliday) [reply rtl/include/eclhelper.hpp:41]: @richardkchapman Yes it will.  But I think that is ok for windows only.

**Pull request merged**